feat: support in memory trace compression

DavePearce opened the pull request:
<!-- CURSOR_SUMMARY -->
> [!NOTE]
> Introduces an in-memory LT trace writer with optional gzip compression (configurable via CLI/config), updates file extensions/paths and tests, and bumps go-corset.
> 
> - **Tracing/Serialization**:
>   - Implement in-memory LT trace writer `LtTraceFile` that builds columns/headers and streams to `OutputStream`.
>   - `ZkTracer.writeToFile()` now writes via `LtTraceFile` and uses `GZIPOutputStream` when filename ends with `.gz`; metadata construction refactored to `buildMetaData`.
>   - `TraceWriter` accepts `traceCompression` flag and chooses `.lt.gz`/`.lt` (and temp) extensions; updates path generation and temp write workflow.
> - **RPC/Config**:
>   - Extend `TracesEndpointConfiguration` and `TracesEndpointCliOptions` with `traceCompression` (default true) and plumb into `GenerateConflatedTracesV2`.
>   - Add logs indicating caching and compression settings.
> - **Tests**:
>   - Update test utilities to generate/expect compressed traces (`.lt.gz`).
> - **CI**:
>   - Bump go-corset to `v1.1.29` in GitHub Action.
> 
> <sup>Written by [Cursor Bugbot](https://cursor.com/dashboard?tab=bugbot) for commit 1270157558c9fc9045e47c357738384f4191ca4e. This will update automatically on new commits. Configure [here](https://cursor.com/dashboard?tab=bugbot).</sup>
<!-- /CURSOR_SUMMARY -->

DavePearce committed:
support in memory trace generation

This alters the way in which trace generation occurs, in that it traces
are always generated fully in memory before being written to disk.  This
allows various possibilities.  For example, we can compress the data
prior to writing to disk.  Or, we can use an alternative format which
requires holistic information of the entire trace before being written
to disk (e.g. it uses some kind of heap, etc).

DavePearce committed:
add plugin cli option for trace compression

Trace compression should be enabled by default, but employs a feature
flag which can be used to disable it.

DavePearce committed:
update go-corset to v1.1.29

This updates go-corset to a version which is compatible with the trace
compression code used by this PR.

DavePearce commented on arithmetization/src/main/java/net/consensys/linea/zktracer/LtTraceFile.java:
```diff
@@ -15,141 +15,297 @@
+      this.name = name;
+      this.bitWidth = bitwidth;
+      this.byteWidth = byteWidth(bitwidth);
+      this.longMax = 1L << bitwidth;
```
This will never happen

letypequividelespoubelles commented on arithmetization/src/main/java/net/consensys/linea/zktracer/ZkTracer.java:
```diff
@@ -131,39 +134,67 @@ public ZkTracer(ChainConfig chain, Map<Long, Hash> historicalBlockHashes) {
+      // Construct (in memory) trace file
+      byte[] bytes = LtTraceFile.toBytes(metadata, trace, modulesToTrace);
+      // Compress file (if requested)
+      if (FilenameUtils.getExtension(filename.toString()).equals("gz")) {
```
I'm not sure, but the extension is not `.lt.gz` ? So it should be `contains` and not èquals` no ?

DavePearce replied:
Well the extension will be just `gz` for any file ending in `gz`. In fact, the plugin using extensions `lt.gz` and `lt.tmp.gz`. So what i've done here just decompresses anything that ends with `gz`. Seems reasonable enough to me

letypequividelespoubelles reviewed (approved):
LGTM but I'd like someone else to review it :)

DavePearce committed:
add logging and minor comments

DavePearce committed:
fix out-of-memory issue

There was an issue due to the maximum size of a Java byte array (~2G).
Specifically, the entire trace was being put into a single byte array,
and this was causing an overflow.  Instead, we just write directly using
output streams.
